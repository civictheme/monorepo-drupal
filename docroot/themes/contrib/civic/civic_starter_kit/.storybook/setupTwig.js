const twigDrupal = require('twig-drupal-filters');

/**
 * Configures and extends a standard twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
