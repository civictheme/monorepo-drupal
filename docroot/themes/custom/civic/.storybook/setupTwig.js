const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');

module.exports.namespaces = {
  atoms: resolve(__dirname, '../', 'components/01-atoms'),
  molecules: resolve(__dirname, '../', 'components/02-molecules'),
  organisms: resolve(__dirname, '../', 'components/03-organisms'),
  templates: resolve(__dirname, '../', 'components/04-templates'),
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
