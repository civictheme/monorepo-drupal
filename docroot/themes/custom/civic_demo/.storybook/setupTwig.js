const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');


module.exports.namespaces = {
  atoms: resolve(__dirname, '../', '.storybook-components/01-atoms'),
  molecules: resolve(__dirname, '../', '.storybook-components/02-molecules'),
  organisms: resolve(__dirname, '../', '.storybook-components/03-organisms'),
  templates: resolve(__dirname, '../', '.storybook-components/04-templates'),
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
