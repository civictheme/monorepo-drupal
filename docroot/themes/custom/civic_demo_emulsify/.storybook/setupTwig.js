const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');
const twigBEM = require('bem-twig-extension');
const twigAddAttributes = require('add-attributes-twig-extension');

module.exports.namespaces = {
  atoms: [
    resolve(__dirname, '../../civic_emulsify', 'components/01-atoms'),
    resolve(__dirname, '../', 'components/01-atoms'),
  ],
  molecules: [
    resolve(__dirname, '../../civic_emulsify', 'components/02-molecules'),
    resolve(__dirname, '../', 'components/02-molecules'),
  ],
  organisms: [
    resolve(__dirname, '../../civic_emulsify', 'components/03-organisms'),
    resolve(__dirname, '../', 'components/03-organisms'),
  ],
  templates: [
    resolve(__dirname, '../../civic_emulsify', 'components/04-templates'),
    resolve(__dirname, '../', 'components/04-templates'),
  ]
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigBEM(twig);
  twigDrupal(twig);
  twigAddAttributes(twig);
  return twig;
};
