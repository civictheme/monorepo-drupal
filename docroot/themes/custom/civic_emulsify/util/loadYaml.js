import { resolve } from 'path';
import { readFileSync } from 'fs';
import { parse } from 'yaml';
import R from 'ramda';

/**
 * Small utility function that loads a yaml file and parses it synchronously.
 * This is intended to make composition cleaner.
 *
 * @param {string} relativePath - relative path to a yaml file that will be loaded and parsed.
 *
 * @returns {string} JavaScript object that results from the yaml parsing of the specified file.
 */
export default function loadYaml(relativePath) {
  const fullPath = resolve(__dirname, relativePath);
  return parse(readFileSync(fullPath, 'utf8'));
}
