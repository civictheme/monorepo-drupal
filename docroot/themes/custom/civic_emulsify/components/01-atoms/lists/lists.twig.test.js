import { join } from 'path';
import { render, Twig } from 'twig-testing-library';

import loadYaml from '../../../util/loadYaml';
import { setupTwig, namespaces } from '../../../.storybook/setupTwig';

setupTwig(Twig);

describe('lists', () => {
  it('can render an dl list', async () => {
    const { container } = await render(
      join(__dirname, 'dl.twig'),
      loadYaml(join(__dirname, 'dl.yml')),
      namespaces,
    );

    expect(container).toMatchSnapshot();
  });

  it('can render an ol list', async () => {
    const { container } = await render(
      join(__dirname, 'ol.twig'),
      loadYaml(join(__dirname, 'ol.yml')),
      namespaces,
    );

    expect(container).toMatchSnapshot();
  });

  it('can render an ul list', async () => {
    const { container } = await render(
      join(__dirname, 'ul.twig'),
      loadYaml(join(__dirname, 'ul.yml')),
      namespaces,
    );

    expect(container).toMatchSnapshot();
  });
});
