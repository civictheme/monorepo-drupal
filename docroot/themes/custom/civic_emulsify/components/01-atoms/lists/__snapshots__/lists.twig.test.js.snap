// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lists can render an dl list 1`] = `
<div>
  


  <dl
    class="dl"
  >
    
            

    <dt
      class="dl-term"
    >
      Definition List
    </dt>
    

    <dd
      class="dl-def"
    >
      A number of connected items or names written or printed consecutively, typically one below the other.
    </dd>
    
          

    <dt
      class="dl-term"
    >
      This is a term.
    </dt>
    

    <dd
      class="dl-def"
    >
      This is the definition of that term, which both live in a 
      <code>
        dl
      </code>
      .
    </dd>
    
          

    <dt
      class="dl-term"
    >
      Here is another term.
    </dt>
    

    <dd
      class="dl-def"
    >
      And it gets a definition too, which is this line.
    </dd>
    
          

    <dt
      class="dl-term"
    >
      Here is one last term.
    </dt>
    

    <dd
      class="dl-def"
    >
      With the final definition.
    </dd>
    
      
  </dl>
  

</div>
`;

exports[`lists can render an ol list 1`] = `
<div>
  


  <ol
    class="ol"
  >
    
            


    <li
      class="list-item"
    >
      
          This is the first item in the ordered list.
  
    </li>
    
          


    <li
      class="list-item"
    >
      
      
      <strong>
        This is the optional label
      </strong>
           And here is the item that goes with the label.
  
    </li>
    
          


    <li
      class="list-item"
    >
      
          Here's the third item.
  
    </li>
    
          


    <li
      class="list-item"
    >
      
          And here's the last item.
  
    </li>
    
      
  </ol>
  

</div>
`;

exports[`lists can render an ul list 1`] = `
<div>
  


  <ul
    class="ul"
  >
    
            


    <li
      class="list-item"
    >
      
          This is the first item in the unordered list.
  
    </li>
    
          


    <li
      class="list-item"
    >
      
      
      <strong>
        This is the optional label
      </strong>
           And here is the item that goes with the label.
  
    </li>
    
          


    <li
      class="list-item"
    >
      
          Here's the third item.
  
    </li>
    
          


    <li
      class="list-item"
    >
      
          And here's the last item.
  
    </li>
    
      
  </ul>
  

</div>
`;
